plugins/hls-class-plugin/src/Ide/Plugin/Class.hs

```diff
@@ -38,16 +41,23 @@ import           Language.Haskell.GHC.ExactPrint.Utils   (rs)
 import           Language.LSP.Server
 import           Language.LSP.Types
 import qualified Language.LSP.Types.Lens                 as J
+import qualified Development.IDE.Core.Shake              as Shake
 
 #if MIN_VERSION_ghc(9,2,0)
 import           GHC.Hs                                  (AnnsModule(AnnsModule))
 import           GHC.Parser.Annotation
 #endif
 
-descriptor :: PluginId -> PluginDescriptor IdeState
-descriptor plId = (defaultPluginDescriptor plId)
+newtype Log = Log T.Text deriving Show
```

Comment:
I think, it would be a good idea to change this to more explicit constructors instead of a generic text type. The idea being, that the structured logging can have added information later, but if it is unstructured using text, we lose this structured info. In particular, I propose to change it to the following Log type:

```suggestion
data Log 
  = LogImplementedMethods [T.Text]
  deriving Show
```
and changing the `Pretty` method to prepend the message we currently have in the log message further down.

Reply:
I also added name of the class to the log.